### PR TITLE
Charging Station Templates - Apply Template - optimizations

### PR DESCRIPTION
**Pull request by ClaudeROSSI**
Performance and Memory Footprint optimization
- Get Charging Station Pagination
- Skip chargers with manual configuration
- Shrink logged data

### CODE REVIEW

File: src/scheduler/tasks/CheckChargingStationTemplateTask.ts [hunk 1]
```diff
@@ -32,39 +32,45 @@ export default class CheckChargingStationTemplateTask extends TenantSchedulerTas
...
-        tenantID: tenant.id,
-        action: ServerAction.UPDATE_CHARGING_STATION_WITH_TEMPLATE,
-        module: MODULE_NAME, method: 'applyTemplateToChargingStations',
-        message: `Bypassed tenant ${Utils.buildTenantName(tenant)})`
+    // Prepare pagination
+    const limit = Constants.BATCH_PAGE_SIZE; // Avoid loading too much data in one shot
+    const sort = null; // No sort
+    let skip = 0;
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
```
Comment (Ramzay) [inline src/scheduler/tasks/CheckChargingStationTemplateTask.ts:40]: ??

File: src/server/ocpp/utils/OCPPUtils.ts [hunk 2]
```diff
@@ -1583,10 +1587,12 @@ export default class OCPPUtils {
...
         module: MODULE_NAME, method: 'enrichChargingStationWithTemplate',
-        message: 'No Template has been found for this Charging Station',
-        detailedMessages: { chargingStation }
+        message: 'No template has been found',
+        detailedMessages: {
+          chargingStationData: LoggingHelper.shrinkChargingStationProperties(chargingStation)
+        }
       });
-      chargingStation.manualConfiguration = true;
+      chargingStation.manualConfiguration = true; // To be clarified! - Why is this changed here???
```
Comment (Ramzay) [inline src/server/ocpp/utils/OCPPUtils.ts:1595]: For the BootNotification ! We save the charging station just after calling this function...
Smells like a hack :)

**Review by Ramzay — APPROVED** [review]

**Comment by ClaudeROSSI** [comment]
Hi,

@OliveGerste : We have a smartcharging test failing from time to time.
We get that strange result once a while. Are you aware of it?

Thanks,
Claude

image

File: src/scheduler/tasks/CheckChargingStationTemplateTask.ts (hunk 1 above)
Comment (ClaudeROSSI) [reply src/scheduler/tasks/CheckChargingStationTemplateTask.ts:40]: I've put in place a pagination. We exit the loop as soon as the page is empty.

File: src/server/ocpp/utils/OCPPUtils.ts (hunk 2 above)
Comment (ClaudeROSSI) [reply src/server/ocpp/utils/OCPPUtils.ts:1595]: :(

**Comment by ClaudeROSSI** [comment]
test results are green after a second run!

**Pull request merged**